src/coprocessor/endpoint.rs

```diff
@@ -302,9 +296,16 @@ impl<E: Engine> Endpoint<E> {
     /// the given `handler_builder`. Finally, it calls the unary request interface of the
     /// `RequestHandler` to process the request and produce a result.
     async fn handle_unary_request_impl(
+        semaphore: Option<Arc<Semaphore>>,
```

Comment:
The `semaphore` is a field of the `Endpoint`, do we need to pass it as a parameter?

Reply:
If we make this async function take `&self`, the future is not 'static and cannot be spanwed to the thread pool.